include/OnnxMlirRuntime.h

```diff
@@ -123,13 +146,21 @@
  *
  * \subsection freeing-tensor-memory Freeing Tensor Memory
  *
- *
- * When creating an OMTensor with a specified tensor array, a user has the option to
- * create a tensor with or without ownership (omTensorCreateWithOwnership).
- *
- * If the ownership flag is set to "false", then a user is responsible for freeing memory until the last use.
- * Otherwise, if the flag is set to "true", then the destruction of the tensor (omTensorDestroy)
- * will also free any associated memory and data buffers.
+ * In general, if a caller creates a tensor object (omTensorCreate), they are
+ * responsible for deallocating the data buffer separately after the tensor is
+ * destroyed. If onnx-mlir creates the tensor (run_main_graph), then the
+ * tensor object owns the data buffer and its freed automatically when the
+ * tensor is destroyed.
```

Comment:
It is a little vague to me: `when it is destroyed`. Call to OMTensorDestroy or destructor of class?

Reply:
To answer the question, technically either. `OMTensorDestroy(tensor)` or `delete tensor` are built so they wind up doing the same thing.

Overall, I was intentionally vague here. `OMTensorListDestroy()` (or `delete tensor_list` indirectly calls `OMTensorDestroy()` which (depending on the tensor's data buffer ownership) may or may not free the data buffer. 

This happens in the main.c example actually. OMTensorDestroy() is never directly called but everything is freed correctly, which I checked with valgrind.